Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp

```diff
@@ -42,13 +42,54 @@
 #endif
 
 #if PLATFORM(IOS_FAMILY) && !PLATFORM(MACCATALYST)
+#import <libkern/OSThermalNotification.h>
+#import <notify.h>
 #import <pal/spi/ios/MobileGestaltSPI.h>
 #endif
 
 namespace WebKit {
 
+static Seconds adjustedTimeoutForThermalState(Seconds timeout)
+{
+#if PLATFORM(IOS_FAMILY) && !PLATFORM(MACCATALYST)
+    static int thermalState = kOSThermalPressureLevelNominal;
+    static dispatch_once_t once = 0L;
+
+    dispatch_once(&once, ^{
```

Comment:
can you use Objective C blocks in C++ files?

Reply:
Yes, it’s an llvm C extension